app.js

```diff
@@ -12,21 +12,27 @@ const models = path.join(lib, 'models.js');
 
 const {Track} = require(models);
 
+const Configfile = require('electron-config');
+const configfile = new Configfile();
+
 app.on('ready', () => {
-  const folders = new Set([app.getPath('music')]);
   const acceptedExtensions = /\.mp3$/;
 
   const window = new BrowserWindow({ title: 'Bityn' });
   window.loadURL('file://' + templates + '/window-main.html');
   window.webContents.on('did-finish-load', function () {
-    const selectedFolders = dialog.showOpenDialog(window, {
-      title: 'Music folder',
-      defaultPath: app.getPath('music'),
-      properties: ['openDirectory']
-    });
-
-    if (selectedFolders !== undefined) {
-      folders.add.apply(folders, selectedFolders);
+    if (!configfile.has('folder')) {
+      var folders = dialog.showOpenDialog(window, {
+        title: 'Music folder',
+        defaultPath: app.getPath('music'),
+        properties: ['openDirectory']
+      });
+      if (folders.indexOf(app.getPath('music')) === -1) {
```

Comment:
This can be skipped if a Set is used. A Set makes sure there are no repetitions.
```javascript
let folders;
if(configFile.has('musicFolders')) {
   folders = configFile.get('folders');
} else {
  folders = new Set([app.getPath('music')]);
  let selectedFolders = dialog.showOpenDialog(window, {
           title: 'Music folder',
           defaultPath: app.getPath('music'),
           properties: ['openDirectory']
  });
  folders.add.apply(folders, selectedFolders);
}
```
This last line does some strange things.
First, it calls "apply" on a method. Apply lets you specify the "this" (the first parameter), in this case **folders**, and then an array for the arguments, in this case **selectedFolders**.
Its the same as doing this: folders.add(folder1, folder2, folder3...)